### DISCUSSION TITLE
Spike PRIME Program Storage

### DISCUSSION THREAD

**Discussion by DylanBPY** [pybricks/support#1851] [Q&A] [answered]
How much space is there on the hub for the program? I see this on the website, is it the user data storage or total storage?

![image](https://github.com/user-attachments/assets/37a6f505-3f2b-4710-9dff-3d240f48c6a4)

**Comment by laurensvalk** [comment, marked as answer]
There is up to `256 KB` reserved for user programs at the moment.

In theory we could allow bigger files, but then your program would not load into RAM. So bigger files are primarily useful for things like sounds, but at the moment we only support beep sounds.
